aten/src/ATen/native/quantized/cuda/fake_quantize_core.cu

```diff
@@ -25,17 +27,22 @@ void fake_quantize_slice_kernel_cuda(
     int64_t quant_min,
     int64_t quant_max) {
   float inv_scale = 1.0f / scale;
-  at::cuda::CUDA_tensor_apply2<float, float>(
-      input, output, [=] __device__(const float& input_val, float& result_val) {
-        result_val = (fminf(
-                          quant_max,
-                          fmaxf(
-                              quant_min,
-                              static_cast<int64_t>(std::nearbyint(
-                                  input_val * inv_scale + zero_point)))) -
-                      zero_point) *
-            scale;
-      });
+  auto iter = TensorIterator();
+  iter.dont_compute_common_dtype();
```

Comment:
I'm fine with not computing common dtype, but I think we should assert here that both input and output are float tensors

Reply:
There is an assert in https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/quantized/fake_quant_per_tensor_affine.cpp#L34

Reply:
Also in per channel as well

Reply:
Yeah, nm, noticed that there's an assert already. Then comment would be nice that we are guaranteed to get only float tensors (also comment above `Returns:
  Fake quantized tensor (double dtype).	  Fake quantized tensor (double dtype).` is inaccurate.

Reply:
fixed